qiskit/primitives/base_estimator.py

```diff
@@ -159,6 +168,28 @@ def __init__(
                         f"expected {circ.num_parameters}, actual {len(params)}."
                     )
 
+    def __new__(
+        cls,
+        circuits: tuple[QuantumCircuit] | QuantumCircuit,
+        observables: tuple[SparsePauliOp] | SparsePauliOp,
+        *args,  # pylint: disable=unused-argument
+        parameters: Iterable[Iterable[Parameter]] | None = None,  # pylint: disable=unused-argument
+        **kwargs,  # pylint: disable=unused-argument
+    ):
+
+        if isinstance(circuits, QuantumCircuit):
+            circuits = (circuits,)
+        else:
+            circuits = tuple(circuits)
+        if isinstance(observables, (PauliSumOp, BaseOperator)):
```

Comment:
By the typehint the `observables` can only be a `SparsePauliOp` (or tuple of it), should the typehint also support `PauliSumOp` and `BaseOperator`?

Reply:
This is a tough story... 
The BaseClass type hint is correct only for SparsePauliOp (and Sequence[SparsePauliOp]). It would be better to switch the case by Sequence or not here, so I will fix it.

Reply:
Done in 0205703.